taskfiles/utils.yml

```diff
@@ -1,64 +1,87 @@
 version: "3"
 
+set: ["u", "pipefail"]
+shopt: ["globstar"]
+
 tasks:
   # ===
   # CHECKSUM UTILS
   # ===
 
-  # @param {string} DATA_DIR The directory to compute the checksum for.
+  # @param {string[]} INCLUDE_PATTERNS Path wildcard patterns to compute the checksum for.
   # @param {string} OUTPUT_FILE
-  # @param {[]string} [EXCLUDE_PATHS] A list of paths, relative to `DATA_DIR`, to exclude from the
-  # checksum.
+  # @param {string[]} [EXCLUDE_PATTERNS] Path wildcard patterns, relative to any `INCLUDE_PATTERNS`,
+  # to exclude from the checksum.
   compute-checksum:
-    desc: "Tries to compute a checksum for the given directory and output it to a file."
+    desc: "Tries to compute a checksum for the given paths and output it to a file."
     internal: true
+    label: "{{.TASK}}-{{.OUTPUT_FILE}}"
     silent: true
     requires:
-      vars: ["DATA_DIR", "OUTPUT_FILE"]
+      vars: ["INCLUDE_PATTERNS", "OUTPUT_FILE"]
     cmds:
+      # We explicitly set --no-anchored and --wildcards to ensure the behaviour for both exclusion
+      # and inclusion matches, as they are only the default for exclusion.
+      # We can quote the exclude patterns for tar because the patterns themselves are passed to tar
+      # as arguments and evaluated by tar during execution. However, for tar --create the input
+      # patterns cannot be quoted as they are evaluated by the shell and the results are passed to
+      # tar as arguments. If the input patterns are passed to tar with quotes the pattern will not
+      # be evaluated and will be treated literally.
```

Comment:
```suggestion
      # We explicitly set `--no-anchored` and `--wildcards` to make the inclusion behaviour match
      # the default exclusion behaviour.
      #
      # We can quote the exclude patterns for `tar` because the patterns themselves are passed to
      # `tar` as arguments and evaluated by `tar` during execution. However, for `tar --create`, the
      # input patterns cannot be quoted since they're evaluated by the shell and the results are
      # passed to `tar` as arguments. If the input patterns are passed to `tar` with quotes, the
      # pattern won't be evaluated and will instead be treated literally.
```